src/Handlebars.Net.Helpers/HandlebarsHelpers.cs

```diff
@@ -1,201 +1,203 @@
-            for (int i = 0; i < parameterCountAllowed.Max() - arguments.Length; i++)
-            {
-                parsedArguments.Add(null);
-            }
-
-            try
-            {
-                return methodInfo.Invoke(obj, parsedArguments.ToArray());
-            }
-            catch (Exception e)
-            {
-                if (e.InnerException != null)
-                {
-                    throw e.InnerException;
-                }
-
-                throw;
-            }
-        }
-
-        /// <summary>
-        /// For Handlebars.Net : also return a helperName surrounded by [ ] to support the official handlebarsjs rules
-        /// For Handlebars.CSharp : only register normal helper name
-        ///
-        /// See https://github.com/StefH/Handlebars.Net.Helpers/issues/7
-        /// </summary>
-        private static string[] CreateHelperNames(string helperName)
-        {
-#if HANDLEBARSNET
-            return new[] { helperName, $"[{helperName}]" };
-#else
-            return new[] { helperName };
-#endif
-        }
-    }
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using HandlebarsDotNet.Helpers.Attributes;
+using HandlebarsDotNet.Helpers.Enums;
+using HandlebarsDotNet.Helpers.Helpers;
+using HandlebarsDotNet.Helpers.Options;
+using HandlebarsDotNet.Helpers.Parsers;
+using HandlebarsDotNet.Helpers.Utils;
+using HandlebarsDotNet.Helpers.Validation;
+
+namespace HandlebarsDotNet.Helpers
+{
+    /// <summary>
+    /// Extension class to Register additional Helpers and BlockHelpers.
+    /// </summary>
+    public static class HandlebarsHelpers
+    {
+        /// <summary>
+        /// Register all (default) or specific categories.
+        /// </summary>
+        /// <param name="handlebarsContext">The <see cref="IHandlebars"/>-context.</param>
+        /// <param name="categories">The categories to register. By default all categories are registered. See the WIKI for details.</param>
+        public static void Register(IHandlebars handlebarsContext, params Category[] categories)
+        {
+            Register(handlebarsContext, o => { o.Categories = categories; });
+        }
+
+        /// <summary>
+        /// Register all (default) or specific categories and use the prefix from the categories.
+        /// </summary>
+        /// <param name="handlebarsContext">The <see cref="IHandlebars"/>-context.</param>
+        /// <param name="optionsCallback">The options callback.</param>
+        public static void Register(IHandlebars handlebarsContext, Action<HandlebarsHelpersOptions> optionsCallback)
+        {
+            Guard.NotNull(optionsCallback, nameof(optionsCallback));
+
+            var options = new HandlebarsHelpersOptions();
+            optionsCallback(options);
+
+            if (options.CultureInfo != null)
+            {
+                handlebarsContext.Configuration.FormatProvider = options.CultureInfo;

```

Comment:
As you're releasing a major version you can introduce some breaking changes. This is a good example: no need to duplicate settings available in Handlebars by default. This can lead to unexpected behavior.